app/models/container_service.rb

```diff
@@ -27,10 +27,23 @@ def container_groups_count
     number_of(:container_groups)
   end
 
+  include EventMixin
   include Metric::CiMixin
 
   PERF_ROLLUP_CHILDREN = :container_groups
 
+  def event_where_clause(assoc = :ems_events)
+    case assoc.to_sym
+    when :ems_events
+      # TODO: improve relationship using the id
+      ["container_namespace = ? AND container_group_name IN (?) AND #{events_table_name(assoc)}.ems_id = ?",
+       container_project.name, container_groups.pluck(:name), ems_id]
```

Comment:
seems to me we can't escape querying for the container_project and container_groups belonging to this service.
I didn't find a way to query multiple associations  in one query - even when I use preload or includes it still executes multiple queries